velox/exec/tests/PrintPlanWithStatsTest.cpp

```diff
@@ -0,0 +1,201 @@
+ * limitations under the License.
+ */
+
+#include "velox/exec/PlanNodeStats.h"
+#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec::test;
+
+using facebook::velox::exec::test::PlanBuilder;
+
+class PrintPlanWithStatsTest : public HiveConnectorTestBase {};
+
+void ensureTaskCompletion(exec::Task* task) {
+  // ASSERT_TRUE requires a function with return type void.
+  ASSERT_TRUE(waitForTaskCompletion(task));
+}
+
+// The test outputs of "innerJoinWithTableScan" and
+// "partialAggregateWithTableScan" reflect the documentation for
+// printPlanWithStats. A failure likely means that the documentation needs an
+// update as well.
+
+TEST_F(PrintPlanWithStatsTest, innerJoinWithTableScan) {
+  const int32_t numSplits = 20;
+  const int32_t numRowsProbe = 1024;
+  const int32_t numRowsBuild = 100;
+  std::vector<RowVectorPtr> leftVectors;
+  leftVectors.reserve(numSplits);
+  auto leftFiles = makeFilePaths(numSplits);
+
+  for (int i = 0; i < numSplits; i++) {
+    auto rowVector = makeRowVector({
+        makeFlatVector<int32_t>(
+            numRowsProbe, [&](auto row) { return row - i * 10; }),
+        makeFlatVector<int64_t>(numRowsProbe, [](auto row) { return row; }),
+    });
+    leftVectors.push_back(rowVector);
+    writeToFile(leftFiles[i]->path, rowVector);
+  }
+  auto probeType = ROW({"c0", "c1"}, {INTEGER(), BIGINT()});
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  // 100 key values in [35, 233] range.
+  auto rightKey = makeFlatVector<int32_t>(
+      numRowsBuild, [](auto row) { return 35 + row * 2; });
+  auto rightVectors = {makeRowVector({
+      rightKey,
+      makeFlatVector<int64_t>(numRowsBuild, [](auto row) { return row; }),
+  })};
+  createDuckDbTable("t", {leftVectors});
+  createDuckDbTable("u", {rightVectors});
+
+  auto buildSide = PlanBuilder(planNodeIdGenerator)
+                       .values(rightVectors)
+                       .project({"c0 AS u_c0", "c1 AS u_c1"})
+                       .planNode();
+  core::PlanNodeId leftScanId;
+  auto op = PlanBuilder(planNodeIdGenerator)
+                .tableScan(probeType)
+                .capturePlanNodeId(leftScanId)
+                .hashJoin(
+                    {"c0"},
+                    {"u_c0"},
+                    buildSide,
+                    "",
+                    {"c0", "c1", "u_c1"},
+                    core::JoinType::kInner)
+                .project({"c0", "c1 + 1", "c1 + u_c1"})
+                .planNode();
+
+  auto task = assertQuery(
+      op,
+      {{leftScanId, leftFiles}},
+      "SELECT t.c0, t.c1 + 1, t.c1 + u.c1 FROM t, u WHERE t.c0 = u.c0");
+
+  ensureTaskCompletion(task.get());
+  ASSERT_THAT(
+      printPlanWithStats(*op, task->taskStats()),
+      ::testing::MatchesRegex(
```

Comment:
I know that I introduced this pattern myself, but I'm thinking that it might be difficult to maintain. When failure occurs, it is hard to tell what's wrong. I wonder if it makes sense to add some helper functions to split the messages into individual lines, check each line and print out which lines do not match.